deep_qa/training/trainer.py

```diff
@@ -282,6 +291,10 @@ def train(self):
         # Then we build the model and compile it.
         logger.info("Building the model")
         self.model = self._build_model()
+        if self.num_gpus > 1:
+            with tensorflow.device("/cpu:0"):
```

Comment:
Why is this necessary?  The device scopes you have inside of `make_parallel` aren't enough?